### PR TITLE
Add generate_surface_of_revolution

### PR DESCRIPTION
**Pull request by isuruf**

### CODE REVIEW

File: meshmode/mesh/generation.py [hunk 1]
```diff
@@ -508,6 +508,76 @@ def generate_icosphere(r: float, order: int, *,
...
+# {{{ generate_cylinder
+
+def generate_cylinder(r: float, order: int, *,
+        uniform_refinement_rounds: int = 0,
+        z_lower: float = 0.0,
+        z_upper: float = 1.0,
+        node_vertex_consistency_tolerance: Optional[Union[float, bool]] = None,
+        unit_nodes: Optional[np.ndarray] = None):
+    """
+    :param r: radius of the sphere.
```
Comment (inducer) [inline meshmode/mesh/generation.py:520]: *sphere* cough

File: meshmode/mesh/generation.py [hunk 2]
```diff
@@ -508,6 +508,76 @@ def generate_icosphere(r: float, order: int, *,
 # }}}
 
 
+# {{{ generate_cylinder
+
+def generate_cylinder(r: float, order: int, *,
```
Comment (inducer) [inline meshmode/mesh/generation.py:513]: Add to module docstring

File: meshmode/mesh/generation.py [hunk 3]
```diff
@@ -508,6 +508,76 @@ def generate_icosphere(r: float, order: int, *,
...
+
+    from meshmode.mesh import Mesh
+    mesh = Mesh(
+        vertices, [grp],
+        node_vertex_consistency_tolerance=node_vertex_consistency_tolerance,
+        is_conforming=True)
+
+    if uniform_refinement_rounds:
+        from meshmode.mesh.refinement import refine_uniformly
+        mesh = refine_uniformly(mesh, uniform_refinement_rounds)
```
Comment (inducer) [inline meshmode/mesh/generation.py:560]: Why is there a need? Wouldn't it be easier to start with the appropriate resolution?

File: meshmode/mesh/generation.py [hunk 4]
```diff
@@ -508,6 +508,76 @@ def generate_icosphere(r: float, order: int, *,
...
+        perform after the initial mesh was created.
+    :param z_lower: lower bound for the z axis.
+    :param z_upper: upper bound for the z axis.
+    :param node_vertex_consistency_tolerance: passed to the
+        :class:`~meshmode.mesh.Mesh` constructor. If *False*, no checks are
+        performed.
+    :param unit_nodes: if given, the unit nodes to use. Must have shape
+        ``(3, nnodes)``.
+    """
+    circle_param = np.linspace(0.0, 1.0, 6, endpoint=False)
```
Comment (inducer) [inline meshmode/mesh/generation.py:533]: Why hardcode 6?

File: meshmode/mesh/generation.py (hunk 1 above)
Comment (inducer) [inline meshmode/mesh/generation.py:520]: Describe the orientation and location of the thing in space.

File: meshmode/mesh/generation.py (hunk 2 above)
Comment (inducer) [inline meshmode/mesh/generation.py:513]: There's a very easy generalization to surfaces of revolution hiding here. Simply take a function `get_radius(height, angle_parameter)`, where the height range is user-specified, and the angle parameter is 0..2pi or 0..1.

**Review by inducer — COMMENTED** [review]
Thanks! A few things below.

File: meshmode/mesh/generation.py (hunk 2 above)
Comment (inducer) [inline meshmode/mesh/generation.py:513]: Oh. Forgot one: add a mini test that runs this. :)

**Comment by isuruf** [comment]
@inducer, this needs your approval to run github actions.

File: meshmode/mesh/generation.py [hunk 5]
```diff
@@ -508,6 +509,78 @@ def generate_icosphere(r: float, order: int, *,
...
+def generate_surface_of_revolution(
+        get_radius: Callable[[np.ndarray, np.ndarray], np.ndarray],
+        height_discr: np.ndarray,
+        angle_discr: np.ndarray,
+        order: int, *,
+        node_vertex_consistency_tolerance: Optional[Union[float, bool]] = None,
+        unit_nodes: Optional[np.ndarray] = None):
+    """
+    :param get_radius: A callable function that takes in an array of heights
+        and an array of angles and returns an array of radii.
```
Comment (inducer) [inline meshmode/mesh/generation.py:523]:
```suggestion
    :param get_radius: A callable function that takes in a 1D array of heights
        and a 1D array of angles and returns a 1D array of radii.
```

File: meshmode/mesh/generation.py [hunk 6]
```diff
@@ -508,6 +509,78 @@ def generate_icosphere(r: float, order: int, *,
...
+# {{ generate_surface_of_revolution
+
+def generate_surface_of_revolution(
+        get_radius: Callable[[np.ndarray, np.ndarray], np.ndarray],
+        height_discr: np.ndarray,
+        angle_discr: np.ndarray,
+        order: int, *,
+        node_vertex_consistency_tolerance: Optional[Union[float, bool]] = None,
+        unit_nodes: Optional[np.ndarray] = None):
+    """
```
Comment (inducer) [inline meshmode/mesh/generation.py:521]:
```suggestion
    """Return a cylinder aligned with the "height" axis aligned with the Z axis.
    
```

**Review by inducer — APPROVED** [review]
Thanks! (I'll apply the doc nits)

**Pull request merged**